### DISCUSSION TITLE
What customizations are available for helm's matching?

### DISCUSSION THREAD

**Discussion by pedz** [emacs-helm/helm#2569] [Q&A]
I tried asking this question on the Emacs Stack Exchange but didn't get any replies.

I've been looking for a list of options that control how `helm` matches and sorts the "source" list.  Here are some examples that I've hit over the past week.

In today's example, I was using `helm-projectile-find-file` looking for `gif.rb` and Helm takes the `.` the same as a space. Rather I needed to type `qif\.rb` to get Helm to narrow its focus like I wanted it. Is there an option that controls this to make dot work as a normal character?

A few days ago, I was working on my Emacs init files which I've put into Org mode files. In that case, when I typed `foo` using `helm-find-files`, the resulting list would include `foo.org`, `foo.el`, and `foo.elc`. I'd like to never see the foo.elc file and I'd prefer to have the `foo.org` be before the `foo.el` file. Indeed, if both of them exist, utopia would be for `foo.el` to be in different "source".

Last, if there is an exact match in the list, I'd like it to be picked first. Very often I don't use Helm like I should (pilot error) and I do `C-x foodog` and whack the return and expect `foodog` to have been selected but sometimes it isn't.  Rather it is Helm's best guess of what it thought I wanted. Helm's history plays a part in this as well.  What is odd in I see in other cases, Helm seems to do this -- pull the exact match up to the top hit.

Thank you for all your help.

**Comment by thierryvolpiatto** [comment]
Perry Smith ***@***.***> writes:

> I tried asking this question on the Emacs Stack Exchange but didn't get any replies.
>
> I've been looking for a list of options that control how helm matches and sorts the "source" list. Here are some examples that I've hit over the past week.
>
> In today's example, I was using helm-projectile-find-file looking for
> gif.rb and Helm takes the . the same as a space. Rather I needed to
> type qif\.rb to get Helm to narrow its focus like I wanted it. Is
> there an option that controls this to make dot work as a normal
> character?

Don't know how helm-projectile-find-file works, but for helm-find-files,
see below.

> A few days ago, I was working on my Emacs init files which I've put
> into Org mode files. In that case, when I typed foo using
> helm-find-files, the resulting list would include foo.org, foo.el, and
> foo.elc. I'd like to never see the foo.elc file and I'd prefer to have
> the foo.org be before the foo.el file. Indeed, if both of them exist,
> utopia would be for foo.el to be in different "source".

In the helm directory I have .el and .elc files, if I want to match
ONLY helm.elc, I add a space and enter "helm\.elc":
/home/me/helm/ helm\.elc

By default helm-find-files is fuzzy matching, it is fine in most cases,
but in such case it is better to regexp match, by adding a space, you
switch to multimatch which is regexp based.

> Last, if there is an exact match in the list, I'd like it to be picked first. Very often I don't use Helm like I should (pilot error) and I do C-x foodog and whack the return and expect
> foodog to have been selected but sometimes it isn't. Rather it is Helm's best guess of what it thought I wanted. Helm's history plays a part in this as well. What is odd in I see in other
> cases, Helm seems to do this -- pull the exact match up to the top hit.
>
> Thank you for all your help.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.*Message ID: ***@***.***>


-- 
Thierry
